pkg/operator/ceph/config/network.go

```diff
@@ -65,12 +67,23 @@ func generateNetworkSettings(context *clusterd.Context, namespace string, networ
 			return []Option{}, errors.Wrapf(err, "failed to get network attachment definition configuration for selector %q", selectorKey)
 		}
 
-		if netConfig.Ipam.Subnet != "" {
-			cephNetworks = append(cephNetworks, configOverride("global", fmt.Sprintf("%s_network", selectorKey), netConfig.Ipam.Subnet))
+		networkRange := getNetworkRange(netConfig)
+		if networkRange != "" {
+			cephNetworks = append(cephNetworks, configOverride("global", fmt.Sprintf("%s_network", selectorKey), networkRange))
 		} else {
 			return []Option{}, errors.Errorf("empty subnet from network attachment definition %q", networkSelectors[selectorKey])
 		}
 	}
 
 	return cephNetworks, nil
 }
+
+func getNetworkRange(netConfig k8sutil.NetworkAttachmentConfig) string {
```

Comment:
Unit test

Reply:
in the unit tests we have two networks, so I changed one to use whereabouts ipam
https://github.com/rook/rook/pull/5864/files#diff-a2cb0276442d401b4bcc407550bfcad0R78
Shall I add a seperate unit test for this function?

Reply:
Yes.

Reply:
Any plans to add some unit tests?

Reply:
Added